### PR TITLE
support for custom executable

### LINKED ISSUE #50: Support for custom shell executable
`subprocess.Popen` with `shell=True` runs the program through `/bin/sh` [by default](https://docs.python.org/3/library/subprocess.html#popen-constructor), a behavior that can be overridden with `executable=value` when creating the object.

Sultan should support this overriding - with a very simple rationale (for me :) - Ubuntu links `/bin/sh` to `/bin/dash` (they probably have their good reasons for that), which does not have the `source` command (it's a bash one). 
Thus the `src` context argument cannot be used, which for python's virtual environments is a bummer.

### PR DESCRIPTION
**Pull request by tminakov**
Adds a new `executable` argument to `load()`, the value of which is the actual shell to use for the execution. Default value is None, preserving the current behavior to run through `/bin/sh` (on POSIX systems).

Fixes #50

### CODE REVIEW

**Comment by davydany** [comment]
@tminakov can you resolve the merge conflicts?

File: docs/sultan-examples.rst [hunk 1]
```diff
@@ -220,4 +220,27 @@ with Sultan.load() as s:
...
+
+    with Sultan.load(executable='/bin/bash') as sultan_bash:
+
+        result = sultan_bash.ps('-p', '$$', '-ocomm=')
+        assert result == 'bash'
+
+    with Sultan.load(executable='/bin/dash') as sultan_other:
+
+        result = sultan_other.ps('-p', '$$', '-ocomm=')
+        assert result == 'dash'
```
Comment (davydany) [inline docs/sultan-examples.rst]: Great docs!

File: src/sultan/api.py [hunk 2]
```diff
@@ -191,14 +196,15 @@ def run(self, halt_on_nonzero=True, quiet=False, q=False):
...
-
+        executable = self.current_context.get('executable')
         try:
             process = subprocess.Popen(commands,
                                        shell=True,
                                        env=env,
                                        stdin=subprocess.PIPE,
                                        stdout=subprocess.PIPE,
                                        stderr=subprocess.PIPE,
+                                       executable=executable,
```
Comment (davydany) [inline src/sultan/api.py]: :+1:

File: test/unit/sultan/test_api.py [hunk 3]
```diff
@@ -250,6 +259,24 @@ def test_calling_context(self):
             if os.path.exists(filepath):
                 os.unlink(filepath)
 
+        # custom executable
+        filehandle, filepath = tempfile.mkstemp()
+        try:
+            with Sultan.load(executable=filepath) as s:
+                self.assertEqual(s.current_context, {
```
Comment (davydany) [inline test/unit/sultan/test_api.py]: :+1:

**Comment by tminakov** [comment]
@davydany I see you took care of it; thanks!

**Pull request merged**